spec/features/user_index_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.feature 'User Index', type: :feature do
+  scenario 'visiting the user index page' do
+    User.create(name: 'Alex', photo: 'https://www.goodfreephotos.com/albums/bolivia/other-bolivia/mountains-and-lake-landscape-scenic.jpg')
+    User.create(name: 'Abdelaziz', photo: 'https://www.goodfreephotos.com/albums/bolivia/other-bolivia/mountains-and-lake-landscape-scenic.jpg')
+
+    visit users_path
+
+    expect(page).to have_content('Alex')
+    expect(page).to have_content('Abdelaziz')
+    expect(page).to have_css("img[alt='Alex']", count: 1)
+    expect(page).to have_css("img[alt='Abdelaziz']", count: 1)
+  end
+
+  scenario 'visiting the user index page, you see the number of posts each user has written..' do
+    user1 = User.create(name: 'Alex')
+    User.create(name: 'Abdelaziz')
+    Post.create(author: user1, title: 'first post')
+    Post.create(author: user1, title: 'second post')
+    Post.create(author: user1, title: 'third post')
+
+    visit users_path
+
+    expect(page).to have_content('3')
+    expect(page).to have_content('0')
+  end
+
+  scenario 'clicking on a user redirects to their show page' do
+    user = User.create(name: 'Salim')
+
+    visit users_path
+
+    click_link 'Salim'
```

Comment:
- [ ] Great effort working on the integration spec project milestone, but currently out of the 14 test cases you have written only 2 are passing and the remaining 12 are failing. I would kindly suggest you work on all the test cases you have written and be sure they are written perfectly and are all passing.
Also, contrary to the number of test cases requested by the student to test for in this milestone which is 28/29 you have only implemented 14 out of the 28/29. I would kindly suggest you write test cases for other requirements too and be sure you have a significant number of test cases written.

Below is a video recording of my claim 👇 

[screen-capture (5).webm](https://github.com/AbdelazizSoliman/Blog-App/assets/78634841/cd1158ca-0b73-4490-a29a-6cc6356f3048)

Reply:
Done, and now I have more test cases, and all passed.